ls1intum/Artemis
# `Development`: Refactor file creation logic

### Checklist
#### General
#### Server

### Motivation and Context
While looking into the issues with the drag and drop file uplaod, @krusche and I noticed that the file service at some places creates empty files, which get deleted later on in the process anyways. This PR aims to simplify the FileService by removing these unnecesary file opearations. 

### Description
I refactored two methods to just return a file path instead of creating actual files. I also used `Path` object at some related places instead of `String` or `File`.

### Steps for Testing
Make sure that uploading files stil works. Please test this on different testservers (especially the staging server) since the DND-issues only occurred there.

Please test that the upload works:
- For DND quiz exercises
- When importing quiz exercises into a course / exam
- Uploading a lecture unit / attachment

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
unchanged
